code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm

```diff
@@ -16,517 +16,23 @@
 /obj/item/weapon/reagent_containers/food/drinks/drinkingglass/on_reagent_change()
 	cut_overlays()
 	if (reagents.reagent_list.len > 0)
```

Comment:
This can be made a lot simpler:

```
/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/on_reagent_change()
	if(reagents.reagent_list.len)
		var/datum/reagent/R = reagents.get_master_reagent()
		icon_state = R.drink_icon
		name = R.drink_name
		desc = R.drink_desc
	else
		icon_state = "glass_empty"
		name = "glass"
		desc = "Your standard drinking glass."
```
Then just have these on the base reagent:

```
	var/drink_icon = "glass_brown"
	var/drink_name = "Glass of ..what?"
	var/drink_desc = "You can't really tell what this is."
```

Then go about copy pasting all the drink icons, names, and descriptions to their unique reagents.